### ISSUE TITLE
Produces an error when prompt would be wrapped in Windows

### ISSUE THREAD

**Issue by Porges** [kkawakam/rustyline#280]
I’m not sure what *should* happen in this situation but it doesn’t feel very graceful 🙂 

When the Windows terminal is sized less than the prompt width, `readline` produces an `Err` result with Windows-specific error code:

![image](https://user-images.githubusercontent.com/12575/63666444-b6ee6c80-c823-11e9-863f-6cb6a0c4fe2a.png)

Repro code:

```rust
extern crate rustyline;

use rustyline::Editor;

fn main() {
    // `()` can be used when no completer is required
    let mut rl = Editor::<()>::new();
    loop {
        let readline = rl.readline("a line long enough to wrap when resized >> ");
        match readline {
            Ok(line) => {
                println!("Line: {}", line);
            },
            Err(err) => {
                println!("Error: {:?}", err);
                break
            }
        }
    }
}
```

The actual problem is with [`SetConsoleCursorPosition`](https://github.com/kkawakam/rustyline/blob/master/src/tty/windows.rs#L277) failing.

**Comment by Porges** [comment]
Some ideas (probably should be configurable):
* Prompt is truncated starting from left
* Prompt is wrapped somehow
* Ability to specify a shorter prompt as an alternative?

**Comment by gwenn** [comment]
Thank you for the detailed report.
I will check and keep you posted.

**Comment by gwenn** [comment]
Should be fixed by #283

**Comment by gwenn** [comment]
Should be fixed by version 5.0.3
